src/parser/statement.js

```diff
@@ -724,8 +726,8 @@ pp.parseClassBody = function (node) {
 
       // disallow static prototype method
       const isStaticPrototype = method.static && (
-        (key.type === "Identifier" && key.name === "prototype") ||
-        (key.type === "StringLiteral" && key.value === "prototype")
+        (key.name === "prototype") || // Identifier
+        (key.value === "prototype")   // Literal
```

Comment:
no need to check the types, just check the name/value. Helps with estree (StringLiteral != Literal)